etl/steps/data/garden/who/2023-08-01/mortality_database.py

```diff
@@ -0,0 +1,173 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+from typing import cast
+
+from owid.catalog import Dataset, Table, Variable, VariableMeta
+from owid.catalog.utils import underscore
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow = cast(Dataset, paths.load_dependency("mortality_database"))
+
+    # Read table from meadow dataset.
+    tb = ds_meadow["mortality_database"]
+    tb = tb.reset_index()
+    #
+    # Process data.
+    #
+    tb = tidy_sex_dimension(tb)
+    tb = tidy_age_dimension(tb)
+    tb = tidy_causes_dimension(tb)
+    tb: Table = geo.harmonize_countries(
+        df=tb, countries_file=paths.country_mapping_path, excluded_countries_file=paths.excluded_countries_path
+    )
+
+    ds_garden = add_metadata(dest_dir=dest_dir, ds_meadow=ds_meadow, tb=tb)
+
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+
+def tidy_causes_dimension(tb: Table) -> Table:
+    """
+    To clarify blood disorders are also included in this group.
+    """
+    cause_dict = {"Diabetes mellitus and endocrine disorders": "Diabetes mellitus, blood and endocrine disorders"}
+    tb["cause"] = tb["cause"].replace(cause_dict, regex=False)
+    return tb
+
+
+def tidy_sex_dimension(tb: Table) -> Table:
+    """
+    Improve the labelling of the sex column
+    """
+    sex_dict = {"All": "Both sexes", "Female": "Females", "Male": "Males", "Unknown": "Unknown sex"}
+    tb["sex"] = tb["sex"].replace(sex_dict, regex=False)
+    return tb
+
+
+def tidy_age_dimension(tb: Table) -> Table:
+    age_dict = {
+        "[Unknown]": "Unknown age",
+        "[85+]": "over 85 years",
+        "[80-84]": "80-84 years",
+        "[75-79]": "75-49 years",
+        "[70-74]": "70-74 years",
+        "[65-69]": "65-69 years",
+        "[60-64]": "60-64 years",
+        "[55-59]": "55-59 years",
+        "[50-54]": "50-54 years",
+        "[45-49]": "45-49 years",
+        "[40-44]": "40-44 years",
+        "[35-39]": "35-39 years",
+        "[30-34]": "30-34 years",
+        "[25-29]": "25-29 years",
+        "[20-24]": "20-24years",
+        "[15-19]": "15-19 years",
+        "[10-14]": "10-14 years",
+        "[5-9]": "5-9 years",
+        "[1-4]": "1-4 years",
+        "[0]": "less than 1 year",
+        "[All]": "all ages",
+    }
+
+    tb["age_group"] = tb["age_group"].replace(age_dict, regex=False)
+
+    return tb
+
+
+def add_metadata(dest_dir: str, ds_meadow: Dataset, tb: Table) -> Dataset:
```

Comment:
this is cool! the only thing is that it wasn't immediately clear to me that you were creating different tables here for each group. Maybe add this note to docstring? e.g.,
`  """ Returns:
   Dataset: containing tables with causes of mortality for each group ("cause", "sex", "age_group")
    """`